### ISSUE TITLE
Support early stopping in xgboost models

### ISSUE THREAD

**Issue by mcloughlin2** [ATOMScience-org/AMPL#315]
When we plot predicted vs actual values for an xgboost model, the model often appears to be overfit to the training set. This could probably be alleviated by training for fewer generations, just as it does for NN models. The current version of the xgboost package allows you to specify a validation dataset and metric and an early stopping rounds parameter, so that you can stop iterations when the metric value for the validation set doesn't improve after early_stopping_rounds iterations.
